Tests/MongoSwiftTests/CodecTests.swift

```diff
@@ -391,17 +380,21 @@ final class CodecTests: MongoSwiftTestCase {
             "ts": Timestamp(timestamp: 1, inc: 2),
             "int32": 5,
             "int64": Int64(6),
-            "dec": Decimal128("1.2E+10"),
+            "dec": Decimal128("1.2E+10")!,
```

Comment:
Why does this need to be force unwrapped? I don't see us making the initializer for Decimal128 return an optional in this PR, did that change?

Reply:
That changed in this PR.

Before this patch, we just stored a `String` within our `Decimal128` type and used libmongoc's string interop for writing to / reading from a document. In order to properly pass the spec tests marked as "lossy" and to properly round trip all `Decimal128`s, we need to store the actual `bson_decimal128_t`.

The old, non-optional initializer didn't bother constructing the `bson_decimal_128_t` since it never stored it. Now, since we always need to store it, the default (and only) String initializer does try to construct one and returns nil if it fails.